viskores/filter/uncertainty/worklet/FiberUncertainUniform.h

```diff
@@ -0,0 +1,531 @@
+{
+class MultiVariateMonteCarlo : public viskores::worklet::WorkletMapField
+{
+public:
+  // Worklet Input
+  // Fiber(const std::vector<std::pair<double, double>>& minAxis,
+  //      const std::vector<std::pair<double, double>>& maxAxis)
+  //  : InputBottomLeft(minAxis), InputTopRight(maxAxis){};
+  MultiVariateMonteCarlo(const viskores::Pair<viskores::Float64, viskores::Float64>& minAxis,
+                         const viskores::Pair<viskores::Float64, viskores::Float64>& maxAxis,
+                         const viskores::Id numSamples)
+    : InputBottomLeft(minAxis)
+    , InputTopRight(maxAxis)
+    , NumSamples(numSamples){};
+
+  // Input and Output Parameters
+  using ControlSignature = void(FieldIn, FieldIn, FieldIn, FieldIn, FieldOut);
+
+  using ExecutionSignature = void(_1, _2, _3, _4, _5);
+  using InputDomain = _1;
+
+  // Template
+  template <typename MinX, typename MaxX, typename MinY, typename MaxY, typename OutCellFieldType>
+  // Operator
+  VISKORES_EXEC void operator()(const MinX& EnsembleMinX,
+                            const MaxX& EnsembleMaxX,
+                            const MinY& EnsembleMinY,
+                            const MaxY& EnsembleMaxY,
+                            OutCellFieldType& probability) const
+  {
+    // User defined rectangle (trait)
+    viskores::FloatDefault minX_user = static_cast<viskores::FloatDefault>(InputBottomLeft.first);
+
+    viskores::FloatDefault minY_user = static_cast<viskores::FloatDefault>(InputBottomLeft.second);
+
+    viskores::FloatDefault maxX_user = static_cast<viskores::FloatDefault>(InputTopRight.first);
+
+    viskores::FloatDefault maxY_user = static_cast<viskores::FloatDefault>(InputTopRight.second);
+
+
+
+    viskores::FloatDefault N1 = 0.0;
+    viskores::FloatDefault N2 = 0.0;
+    viskores::Id NonZeroCases = 0;
+    viskores::FloatDefault MCProbability = 0.0;
+
+    // Data rectangle
+    viskores::FloatDefault minX_dataset = static_cast<viskores::FloatDefault>(EnsembleMinX);
+    viskores::FloatDefault maxX_dataset = static_cast<viskores::FloatDefault>(EnsembleMaxX);
+    viskores::FloatDefault minY_dataset = static_cast<viskores::FloatDefault>(EnsembleMinY);
+    viskores::FloatDefault maxY_dataset = static_cast<viskores::FloatDefault>(EnsembleMaxY);
+
+
+    viskores::FloatDefault minX_intersection = std::max(minX_user, minX_dataset);
+    viskores::FloatDefault minY_intersection = std::max(minY_user, minY_dataset);
+    viskores::FloatDefault maxX_intersection = std::min(maxX_user, maxX_dataset);
+    viskores::FloatDefault maxY_intersection = std::min(maxY_user, maxY_dataset);
+
+
+// Monte Carlo
+// Trait Coordinates (minX_user,minY_user) & (maxX_user,maxY_user)
+#if defined(VISKORES_CUDA) || defined(VISKORES_KOKKOS_HIP)
+    thrust::minstd_rand rng;
+    thrust::uniform_real_distribution<viskores::FloatDefault> distX(minX_dataset, maxX_dataset);
+    thrust::uniform_real_distribution<viskores::FloatDefault> distY(minY_dataset, maxY_dataset);
```

Comment:
To use `ArrayHandleRandomUniformReal`, you would create the array in the control environment and then pass it to the worklet as a `WholeArrayIn`.